cmd/pipeline/deploy.go

```diff
@@ -58,7 +57,7 @@ func deploy(ctx context.Context) *cobra.Command {
 
 			var err error
 			if name == "" {
-				name, err = getPipelineName()
+				name, err = getPipelineName(repository)
```

Comment:
Where are we infering the repo? could we infer the repo before this point?
that way, we will always infer the name from the repo